### ISSUE TITLE
Building Failed - java installed

### ISSUE THREAD

**Issue by p01t0t0** [AhMyth/AhMyth-Android-RAT#51]
Trying to build APK with no luck, i always get "Building Failed".
Java is already installed.

**Comment by p01t0t0** [comment]
After a few hours of struggling, i decided to reinstall my OS - kali light
Gave another try with AhMyth and now it can build the APK.
Using the following steps to install java and it is working good now.
https://computingforgeeks.com/how-to-install-latest-java-jdk-on-debian-8-kali-linux-2-0-and-ubuntu-15-04/

**Comment by p01t0t0** [comment]
The app is now working, no more issue generating the APK file.
The app is working good on my local network.
Jut one issue, i cannot bind the generated APK file with another app.
Can anyone help!

**Comment by nehatkkhan786** [comment]
Brother is there any solution that without reinstalling the OS we can fix the 'building the apk failed'  error!

**Comment by p01t0t0** [comment]
@nehatkkhan786, did you install java?

If not, follow steps on blog below to install it.
https://computingforgeeks.com/how-to-install-latest-java-jdk-on-debian-8-kali-linux-2-0-and-ubuntu-15-04/

If you already installed it, then just uninstall it and re-install it with step above.

That's what worked for me!!!!!!

**Comment by nehatkkhan786** [comment]
yes brother i did that but nothing works its still showing that (Adding source ip:port Failed).

**Comment by p01t0t0** [comment]
@nehatkkhan786,

Check [this issue](https://github.com/AhMyth/AhMyth-Android-RAT/issues/17).

**Comment by burhanuday** [comment]
Download the code and open the apk folder in Android Studio or Eclipse, Go to iosocket.java file and change the ip address to your ip address and compile (i did this using Android Studio 3.0 and it is working for me) @p01t0t0

**Comment by nehatkkhan786** [comment]
brother i have not find any apk folder in the source code. can you please tell me the process if you dont mind. @burhanuday

**Comment by burhanuday** [comment]
@nehatkkhan786 

1. [Download the code by pressing the download button](https://github.com/AhMyth/AhMyth-Android-RAT)

2. Open Android Studio

3. Press Open existing Eclipse project, in the window, select the Ahmyth-Client folder
Wait for gradle build to complete. Install any required build tools and upate project gradle to the latest version when the popup comes up

4. In the IOSocket.java file, find the ip address and port line and change it to your global ip address and the port you have opened on your router

5. Now sign and compile the project, you will have your apk ready

**Comment by Zerukhan123** [comment]
@burhanuday 
You seem like a nice person and it looks like you know a lot about this. I have downloaded Android Studio and have edited the IOSocket.java file but I am struggling compiling the .apk file. I don't know how to do it. Please clarify from step 3 on wards. I shall be very thankful for your help. I am looking forward to you reply !!!!

**Comment by Victor-Frank** [comment]
building failed on parrot os plz help!!!!

**Comment by DiKwickley** [comment]
Tried everything..
building failed everytime..
i am using  java jdk 8u.201...
Please help!
if another versioin of java has to installed please give the full name..
thanks!

**Comment by nominomoses** [comment]
Hey...found a solution for you...open terminal type ..
update-alternatives --config java..
pick the /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java which can be 3 or which ever number yours will be the press enter and try the build again...mine worked

**Comment by Joniuton** [comment]
Thanks nominomoses , it worked out right here!

**Comment by sachdeva-dotcom** [comment]
hey nominomoses
can you describe its all process i am a beginner
pls help

**Comment by dx4iot** [comment]
hey @sachdeva-dotcom 
Open terminal and type:
`update-alternatives --config`
**select 2** and then press enter

**Comment by natiethiopia** [comment]
Trying to bind with the original apk, i always get "Building Failed".
Java is already installed.
Please help

**Comment by P1n3pple** [comment]
> After a few hours of struggling, i decided to reinstall my OS - kali light
> Gave another try with AhMyth and now it can build the APK.
> Using the following steps to install java and it is working good now.
> https://computingforgeeks.com/how-to-install-latest-java-jdk-on-debian-8-kali-linux-2-0-and-ubuntu-15-04/

I use a windows OS, is this still applicable?

**Comment by P1n3pple** [comment]
> @nehatkkhan786, did you install java?
> 
> If not, follow steps on blog below to install it.
> https://computingforgeeks.com/how-to-install-latest-java-jdk-on-debian-8-kali-linux-2-0-and-ubuntu-15-04/
> 
> If you already installed it, then just uninstall it and re-install it with step above.
> 
> That's what worked for me!!!!!!


> @nehatkkhan786, did you install java?
> 
> If not, follow steps on blog below to install it.
> https://computingforgeeks.com/how-to-install-latest-java-jdk-on-debian-8-kali-linux-2-0-and-ubuntu-15-04/
> 
> If you already installed it, then just uninstall it and re-install it with step above.
> 
> That's what worked for me!!!!!!

My OS is windows, what applies to windows?

**Comment by Zanderkail** [comment]
im on windows and i tried the java 8 update but im still getting the build failed error

**Comment by Morsmalleo** [comment]
> im on windows and i tried the java 8 update but im still getting the build failed error

I have fixed this problem in my fork of AhMyth, you also dont need Java 8 anymore with my fork of AhMyth as my fork of AhMyth now runs with Java 11 openjdk, I released version 1.0-beta.2 on my fork of AhMyth as it is much more stable than version 1.0-beta.1, you can download the AhMyth setup.exe file for Windows from the link below, it will tale you to the newest AhMyth release of v1.0-beta.2 on my fork of AhMyth

https://github.com/Morsmalleo/AhMyth/releases/tag/v1.0-beta.2

Just follow the Windows installation instructions in the README.md on my fork of AhMyth
